src/Model/PhasorDynamics/Bus/BusSignal.hpp

```diff
@@ -0,0 +1,129 @@
+
+#pragma once
+
+#include <Model/PhasorDynamics/BusBase.hpp>
+
+namespace GridKit
+{
+  namespace PhasorDynamics
+  {
+    
+    // TODO change base class so there is no Vr and Vi
+
+    template <class ScalarT, typename IdxT>
+    class BusSignal : public BusBase<ScalarT, IdxT>
+    {
+      using BusBase<ScalarT, IdxT>::size_;
+      using BusBase<ScalarT, IdxT>::y_;
+      using BusBase<ScalarT, IdxT>::yp_;
+      using BusBase<ScalarT, IdxT>::yB_;
+      using BusBase<ScalarT, IdxT>::ypB_;
+      using BusBase<ScalarT, IdxT>::f_;
+      using BusBase<ScalarT, IdxT>::fB_;
+      using BusBase<ScalarT, IdxT>::tag_;
+
+    public:
+
+      using real_type = typename BusBase<ScalarT, IdxT>::real_type;
+      using DataT     = BusData<real_type, IdxT>;
+
+      BusSignal(const DataT& data);
+      
+      ~BusSignal()
+      {
+      }
+
+      virtual int allocate() override;
+      virtual int tagDifferentiable() override;
+      virtual int initialize() override;
+      virtual int evaluateResidual() override;
+      virtual int evaluateJacobian() override;
+
+       /**
+       * @brief A one-time initialization function
+       * that can be called by any.
+       * 
+       * @return state value of signal
+       */
+      void initial_value(ScalarT value) override
+      {
+
+        y_[0] = value;
+        //if(is_initialized_)
+        //{
+        //  std::cout << "ERROR!";
+        //}
+        //else{
+        //  y_[0] = value;
+        //  is_initialized_ = true;
+        //}
+      }
+
+       /**
+       * @brief A read only accessor to the signal state
+       * 
+       * @return state value of signal
+       */
+      ScalarT& read() override
+      {
+        return y_[0];
+      }
+
+       /**
+       * @brief A write operation only available
+       * to the source signal
+       * 
+       * @param value The value of the signal
+       */
+      void send(ScalarT& value) override
+      {
+        f_[0] += value;
+      }
```

Comment:
This should be assignment rather than assignment addition:
```suggestion
      void send(ScalarT& value) override
      {
        f_[0] = value;
      }
```
There is only one source that sends the signal.

Reply:
We made the decision in the group discussion that we would have the signal objects own the variables associated with their values. Based on this paradigm, the residual equation for that value would be `f[i] = <calculated value> - x[i]`.  Following the pattern of the electrical buses, the residual calculation of the signal bus first sets the residual equal to `-x[i]`. Then the writing component should add in the calculated value. That is why it must be an addition. 

However, it is important that this function is not called more than once in a given residual calculation. In a future PR, we should add a flag internal to the signal bus to throw an exception if this occurs.

Reply:
We can change this in a later PR, but right now it is needed for the reasons @abirchfield discussed.